python/cudf/cudf/core/series.py

```diff
@@ -453,31 +452,20 @@ def empty(self):
         return not len(self)
 
     def __getitem__(self, arg):
-        data = self._column[arg]
-        index = self.index.take(arg)
-        if is_scalar(data) or data is None:
-            return data
-        return self._copy_construct(data=data, index=index)
-
-    def __setitem__(self, key, value):
-        # coerce value into a scalar or column
-        if is_scalar(value):
-            value = to_cudf_compatible_scalar(value)
+        if isinstance(arg, (slice, range, ColumnBase)) or (
+            pd.api.types.is_bool_dtype(as_column(arg).dtype)
+        ):
+            return self.iloc[arg]
         else:
-            value = column.as_column(value)
+            return self.loc[arg]
 
-        if hasattr(value, "dtype") and pd.api.types.is_numeric_dtype(
-            value.dtype
+    def __setitem__(self, key, value):
+        if isinstance(key, (slice, range, ColumnBase)) or (
+            pd.api.types.is_bool_dtype(as_column(key).dtype)
```

Comment:
Same here

Reply:
loc doesn't handle boolean mask, we may have to introduce this to loc.

Reply:
IIRC in Pandas `iloc` doesn't handle boolean mask but `loc` does, so we should really handle it in both if possible.